linfa-clustering/src/dbscan/hyperparameters.rs

```diff
@@ -39,6 +39,8 @@ impl DbscanHyperParams {
     ///
     /// Defaults are provided if the optional parameters are not specified:
     /// * `tolerance = 1e-4`
+    // Violates the convention that new should return a value of type `Self`
```

Comment:
I would add a `IntoValidatedParams` trait, and have a single structure describing the parameters. The receiver function can then call `validate` of `IntoValidatedParams`, or just panics in the setter functions. But I think as well, that we should add exceptions here for now